src/plugins/intel_cpu/src/memory_state.cpp

```diff
@@ -14,60 +14,51 @@ using namespace InferenceEngine;
-                                                     MemoryDescPtr external_desc,
-                                                     MemoryCPtr init_val) :
-    IVariableState{name}, m_external_desc{external_desc} {
-    reset_prime_mem(mem_build());
-    reset_second_mem(mem_build());
-    m_internal_desc = prime_mem()->getDescPtr();
-    auto&& shape = m_internal_desc->getShape();
-    //TODO what if by some reason we already have internal static state while the node is dynamic, is it even possible?
+VariableStateBase::VariableStateBase(const std::string& name, const MemoryDescPtr& external_desc) :
+    IVariableState{name} , m_external_desc{external_desc} {}
 
-    if (shape.isStatic()) {
-        if (init_val) {
-            prime_mem()->load(*init_val);
-        } else {
-            prime_mem()->nullify();
-        }
-    } else {
-        //in the case of the original desc has dynamic shape we create an empty tensor
-        auto new_desc = to_static(m_internal_desc);
-        prime_mem()->redefineDesc(new_desc);
+MemoryDescPtr VariableStateBase::to_static(const MemoryDescPtr& desc) {
+    if (!desc->isDefined()) {
+        auto&& current_dims = desc->getShape().getDims();
+        VectorDims new_dims(current_dims.size());
+        std::transform(current_dims.begin(), current_dims.end(), new_dims.begin(), [](Dim x) {
+            return x == Shape::UNDEFINED_DIM ? 0 : x; });
+
+        return desc->cloneWithNewDims(new_dims, true);
     }
+    return desc;
 }
 
-void VariableStateDoubleBuffer::set_state(const ov::SoPtr<ov::ITensor>& state) {
+const dnnl::engine& VariableStateBase::get_engine() {
+    static const dnnl::engine eng(dnnl::engine::kind::cpu, 0);
+    return eng;
+}
+
+void VariableStateBase::set_state(const ov::SoPtr<ov::ITensor>& state) {
     m_state = state; // simply to extend the lifetime
     auto state_desc = MemoryDescUtils::generateCpuBlockedMemoryDesc(m_state);
 
     const auto& shape = state_desc->getShape();
 
-    if (prime_mem()->getShape() != shape) {
-        auto new_desc = m_internal_desc->cloneWithNewDims(shape.getStaticDims());
-        prime_mem()->redefineDesc(new_desc);
+    if (input_mem()->getShape() != shape) {
+        auto new_desc = internal_desc()->cloneWithNewDims(shape.getStaticDims());
+        input_mem()->redefineDesc(new_desc);
     }
 
     auto src = m_state->data();
 
     Memory mem(get_engine(), state_desc, src);
-    prime_mem()->load(mem);
+    input_mem()->load(mem);
 }
 
-const dnnl::engine& VariableStateDoubleBuffer::get_engine() const {
-    static const dnnl::engine eng(dnnl::engine::kind::cpu, 0);
-    return eng;
-}
-
-ov::SoPtr<ov::ITensor> VariableStateDoubleBuffer::get_state() const {
-    //TODO , in general case must be synchronized
-    const auto& current_dims = prime_mem()->getStaticDims();
+ov::SoPtr<ov::ITensor> VariableStateBase::get_state() const {
+    //TODO , input_mem name is rather confusing in this context
```

Comment:
Add `prime_mem()` accessor to the base class so that not to confuse with `input_mem`